ocaml/sdk-gen/go/gen_go_helper.ml

```diff
@@ -45,35 +45,57 @@ let generate_file rendered output_file =
     (fun () -> output_string out_chan rendered)
     ~finally:(fun () -> close_out out_chan)
 
+module EnumSet = Set.Make (struct
+  type t = string * (string * string) list
+
+  let compare (x0, _y0) (x1, _y1) = String.compare x0 x1
+end)
```

Comment:
The base type is important, it's better if it's encoded.

This can be done with a simple type definition, or even giving a name to the module here.

The type along the comparison makes it suspect, it looks an awful lot like a key-value type. At that point it makes sense to use a Map:

```suggestion
type enum = (string * string) list

module StringMap = Map.Make (String)

type enums = enum StringMap.t

let choose_enum _key a _b = Some a
```
This mean changing the functions further down: